kong/core/cluster.lua

```diff
@@ -25,25 +26,29 @@ local function async_autojoin(premature)
   -- If this node is the only node in the cluster, but other nodes are present, then try to join them
   -- This usually happens when two nodes are started very fast, and the first node didn't write his
   -- information into the datastore yet. When the second node starts up, there is nothing to join yet.
-  local lock = resty_lock:new("cluster_autojoin_locks", {
+  local lock, err = resty_lock:new("cluster_autojoin_locks", {
     exptime = ASYNC_AUTOJOIN_INTERVAL - 0.001
   })
+  if not lock then
+    ngx_log(ngx.ERR, "failed to init lock dictionary", err)
```

Comment:
The `err` variable will tell what the error is. Should be `"could not create lock: "`